src/test/java/uk/gov/pay/api/it/AgreementsIT.java

```diff
@@ -58,4 +58,45 @@ public void getAgreement_MissingAgreementShouldMapException() {
                 .body("code", is("P0200"))
                 .body("description", is("Not found"));
     }
+
+    @Test
+    public void searchAgreementsFromLedger() {
+        var fixture1 = anAgreementFromLedgerFixture()
+                .withExternalId(agreementId)
+                .build();
+
+        var fixture2 = anAgreementFromLedgerFixture()
+                .withExternalId("another-agreement-id")
+                .withServiceId("service-2")
+                .withReference("ref-2")
+                .withDescription("Description 2")
+                .withStatus("CANCELLED")
+                .withCreatedDate("2022-07-27T12:30:00Z")
+                .build();
+
+        ledgerMockClient.respondWithSearchAgreements(GATEWAY_ACCOUNT_ID, fixture1, fixture2);
+
+        given().port(app.getLocalPort())
+                .header(AUTHORIZATION, "Bearer " + PaymentResourceITestBase.API_KEY)
+                .get(AGREEMENTS_PATH)
```

Comment:
Could we add something like page number and status to the query string just to assert that they get propagated all the way through from the call to the resource out to the wiremock ledger call? 

Even though it won't actually be filtering data I suppose it asserts that the optional params are correctly put into the hash map

Reply:
Done this.